flexmeasures/data/schemas/scheduling/__init__.py

```diff
@@ -211,3 +211,43 @@ def _get_variable_quantity_unit(
                 f"Unexpected type '{type(variable_quantity)}' for '{field}': {variable_quantity}."
             )
         return unit
+
+
+class DBFlexContextSchema(FlexContextSchema):
+
+    @validates_schema
+    def forbid_time_series_specs(self, data: dict, **kwargs):
+        """Do not allow time series specs for the flex-context fields saved in the db."""
+
+        # List of keys to check for time series specs
+        keys_to_check = [
+            "ems_power_capacity_in_mw",
+            "ems_production_capacity_in_mw",
+            "ems_consumption_capacity_in_mw",
+            "ems_consumption_breach_price",
+            "ems_production_breach_price",
+            "ems_peak_consumption_in_mw",
+            "ems_peak_consumption_price",
+            "ems_peak_production_in_mw",
+            "ems_peak_production_price",
+        ]
+
+        # Check each key and raise a ValidationError if it's a list
+        for key in keys_to_check:
+            if key in data and isinstance(data[key], list):
+                raise ValidationError(
+                    f"Time series specs are not allowed in flex-context fields in the DB for '{key}'."
+                )
+
+    @validates_schema
+    def forbid_fixed_prices(self, data: dict, **kwargs):
+        """Do not allow fixed consumption price or fixed production price in the flex-context fields saved in the db."""
```

Comment:
I missed why this is actually needed. Why not set a price in the DB?

In some scenarios, these prices might be constant.

Reply:
A purely effort-based suggestion from my side. Unfortunately there is some tech debt that would need to be paid off to support setting fixed quantities for these two fields. I suggested we deal with that in a separate issue. Note that this exclusion here is without loss of functionality, because the UI also only supported setting a sensor reference before.